### DISCUSSION TITLE
Software work time

### DISCUSSION THREAD

**Discussion by QuinzyW** [openPupil/Open-PupilEXT#9] [Q&A] [answered]
Hi, Thanks for your help, we can get accurate pupil data at a low cost. But I meet a question that the software works at Stereo model only for one hour and the camera screen go to be stuck. We have to restart the software. So is it a common question or just my fault?
We use the PupilEXT with OS: win10, CPU: i51400, memory: 16G.

**Comment by BZandi** [comment, marked as answer]
We had already tested whether long time recordings over a whole day would cause any issue but could not detect any problems in our setting, so this is no expected behaviour. However, it must be noted that with continuous stereo long-term recordings (> 60 min), several factors come into play that could make instability possible. If such an error only occurs after an hour of continuous CSV recording, then the following source of errors can be possible from my point of view, which need to be checked for isolating this issue:

1. When shooting continuously, the cameras can overheat and stop responding. This could be an issue with low-cost cameras. Are passive heat sinks used for the camera? Which camera model series do you use? Can you check the temperature of the cameras ? What is the adjusted frame rate ?

2. if the continuous recording is done in stereo mode, it can lead to memory overflow if the fps is set faster than the computer can handle in real-time. Can you notice in the resource manager of windows a higher memory usage during recording time? Further, we provide the fps count of the camera's image feed and the fps count of the image processing of PupilEXT in the live viewer. Are these numbers at the same level during the whole recording time? Does this issue also appear when you only track the pupil but without CSV recording or frame recording? Does this issue also appear when recording only in a single camera modus? Please check.

3. Maybe the duration of the hardware trigger is not set long enough at the beginning in the camera preference window. Did you make sure that the trigger runtime is set long enough at the beginning? 

4. It can also be possible that your microcontroller gets stuck and does not deliver a hardware trigger to the cameras anymore, resulting in a frozen image. Which microcontroller is used? Did you check whether the microcontroller outputs a stable hardware trigger output even after one hour? Are the microcontroller and the two cameras on the same USB bus?

**Reply by QuinzyW** [reply]
Thanks for your help. I will check the above points.

**Reply by QuinzyW** [reply]
There are two main questions I met
1. Haven't set the hardware trigger long enough.
2. Should open the PupilEXT with `administrator` right.
Thanks for your help, I have solved the question.
